lib/compat/wordpress-trunk/widget-render-api-endpoint.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * Polyfill API class to render widgets via the REST API.
+ *
+ * @see \WP_REST_Controller
+ */
+class WP_REST_Widget_Render_Endpoint_Polyfill extends \WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'widget-types';
+		$this->full_endpoint_instance = new WP_REST_Widget_Types_Controller();
+	}
+
+	/**
+	 * Registers the widget type routes.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		$route = '/' . $this->rest_base . '/(?P<id>[a-zA-Z0-9_-]+)/render';
+
+		// Don't override if already registered
+		if ( rest_get_server()->get_route_options( $route ) ) {
```

Comment:
Is there any better way of checking if the route is already registered?

Reply:
AFAIK, this only works after `WP_REST_Server::get_routes` has been called which initializes the route options. I'm not sure if this would have been done at this point. It would probably be safer to check `rest_get_server()→get_routes( 'wp/v2' )`